pkg/gardenlet/apis/config/types.go

```diff
@@ -336,12 +336,22 @@ type GardenLoki struct {
 	Priority *int32
 }
 
+// ShootNodeLogging contains configuration for the shoot node logging.
+type ShootNodeLogging struct {
+	// ShootPurposes denote which shoots can have node loggin by they purpose
+	ShootPurposes []string
+	// AllowAllExecptTesting allows all shoot of all purpose, except testing to have shoot node logging
+	AllowAllExecptTesting bool
```

Comment:
Why is this needed? We already have `ShootPurposes` which should be good enough.

Reply:
I have used it to not specify all purposes explicitly. But I will remove it.